src/Functions/FunctionsHashing.h

```diff
@@ -183,6 +190,10 @@ struct HalfMD5Impl
 
     static UInt64 combineHashes(UInt64 h1, UInt64 h2)
     {
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+        h1 = Poco::ByteOrder::flipBytes(static_cast<Poco::UInt64>(h1));
```

Comment:
We just enabled compilation of ClickHouse with C++23 (#47424). Could you try to use C++23 `std::byteswap` instead of `flipBytes()`?

Reply:
Done.